src/material-experimental/mdc-button/fab.ts

```diff
@@ -49,27 +63,76 @@ export class MatFabButton extends MatButtonBase {
   // The FAB by default has its color set to accent.
   color = 'accent' as ThemePalette;
 
+  private _extended: boolean;
+  get extended() { return this._extended; }
+  set extended(extended: any) { this._extended = coerceBooleanProperty(extended); }
+
   constructor(
       elementRef: ElementRef, platform: Platform, ngZone: NgZone,
       @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode?: string) {
     super(elementRef, platform, ngZone, animationMode);
   }
 }
 
+/**
+ * Material Design mini floating action button (FAB) component. These buttons represent the primary
+ * or most common action for users to interact with.
+ * See https://material.io/components/buttons-floating-action-button/
+ */
+@Component({
+  selector: `button[mat-mini-fab]`,
+  templateUrl: 'button.html',
```

Comment:
So, a thought occurred to me that having multiple components with the same template would lead to duplicate generated code. I checked with Andrew K, and he confirmed. So, rather than introducing a separate component with the same template, we should probably try to have one class with the `templateUrl` as the base class and then extend all the other buttons from it

Reply:
I believe all buttons extend MatButtonBase with the same template (and anchos MatAnchorBase). Should there be another MatFabBase?